c10/core/TensorImpl.h

```diff
@@ -748,7 +764,7 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
 #ifdef DEBUG
     TORCH_INTERNAL_ASSERT(compute_numel() == numel_);
 #endif
-    return numel_;
+    return numel_.expect_int();
```

Comment:
This should be unchecked, if you have symbolic sizes you never call into the default impl